server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java

```diff
@@ -1260,6 +1338,74 @@ public void testOrderByPipelineAggregation() throws Exception {
         }
     }
 
+    public void testSingleValuedFieldWithScript() throws IOException {
+        try (Directory directory = newDirectory()) {
+            try (RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory)) {
+
+                Document document = new Document();
+                document.add(new NumericDocValuesField("value", 100));
+                indexWriter.addDocument(document);
+
+                try (IndexReader indexReader = maybeWrapReaderEs(indexWriter.getReader())) {
+
+                    MappedFieldType fieldType1 = new NumberFieldMapper.NumberFieldType(NumberFieldMapper.NumberType.LONG);
+                    fieldType1.setName("value");
+                    fieldType1.setHasDocValues(true);
+
+                    IndexSearcher indexSearcher = newIndexSearcher(indexReader);
+
+                    TermsAggregationBuilder aggregationBuilder = new TermsAggregationBuilder("_name", ValueType.NUMERIC)
+                        .field("value")
+                        .script(new Script(ScriptType.INLINE, MockScriptEngine.NAME, SINGLE_SCRIPT, Collections.emptyMap()));
```

Comment:
I think this test is also essentially testing a value script like one below because it's setting a field on the agg builder. From what I remember whether you set the field determines whether the agg gets a `ValuesSource.Numeric.WithScript` or a `ValuesSource.Numeric.Script`

Not sure if this is meaningful to what our testing goals here, but some of the other aggregator tests make this distinction

Reply:
Oops, no that was a mistake.  Did not want the field to be included in this test.  Good catch, thanks! :+1: